tools/generator/GenBaseSupport.cs

```diff
@@ -302,12 +302,14 @@ public override bool ValidateNamespace ()
 			return true;
 		}
 
-		bool IsObfuscatedName (String name)
+		bool IsObfuscatedName (int threshold, string name)
 		{
 			if (name.StartsWith ("R.", StringComparison.Ordinal))
 				return false;
 			int idx = name.LastIndexOf ('.');
 			string last = idx < 0 ? name : name.Substring (idx + 1);
+			if (threshold <= Math.Pow (26, last.Length - 1))
```

Comment:
I don't understand the logic/rationale here, as `26**(last.Length-1)` is possibly *huge*.

If `last` is `c`, then it's `26**0` or `1`, so unless `threshold` is 0, *all* types with names of one character will be allowed. Perhaps that's desirable?

If `last` is `ex`, then it's `26**1` or `26`, so a package would need 27 types for `ex` to be considered obfuscated. Huh?

If `last` is `bitmap`, then it's `26**5`, or `11881376`, at which point it's fair to say that *no* types will be considered to be obfuscated, as `threshold` will *always* be less than `11881376`.

> we just count the sibling nodes and if name length is short enough to fit within the number of classes, we mark as obfucated.

I'm not sure that's what this code does -- though I must admit I'm *very* confused by it -- and if it does do that, I'm not sure that's necessarily useful?

Perhaps a better obfuscation would be "all class names entirely lowercase that are <= 2 characters in length." Do we know of any obfuscated types that are longer than two characters?